db/range_tombstone_fragmenter.h

```diff
@@ -0,0 +1,77 @@
+//  Copyright (c) 2018-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <vector>
+#include <list>
+#include <string>
+#include <memory>
+
+#include "rocksdb/status.h"
+#include "db/dbformat.h"
+#include "db/pinned_iterators_manager.h"
+#include "table/internal_iterator.h"
+
+namespace rocksdb {
+
+SequenceNumber MaxCoveringTombstoneSeqnum(InternalIterator* tombstone_iter,
+                                          const Slice& key,
+                                          const Comparator* ucmp);
+
+// FragmentedRangeTombstoneIterator converts an InternalIterator of a range-del
+// meta block into an iterator over non-overlapping tombstone fragments. The
+// tombstone fragmentation process should be more efficient than the range
+// tombstone collapsing algorithm in RangeDelAggregator because this leverages
+// the internal key ordering already provided by the input iterator. If there
+// are few overlaps, creating a FragmentedRangeTombstoneIterator should be
+// O(n), while the RangeDelAggregator tombstone collapsing is always O(n log n).
+class FragmentedRangeTombstoneIterator : public InternalIterator {
```

Comment:
A note about making this an `InternalIterator`: in the follow-up work I'm planning, SSTs will cache their `FragmentedRangeTombstoneIterator`s, since their contents won't change (though this will require some additional logic to disable snapshot filtering and the dropping of covered fragments). These fragmented iterators will be used everywhere a range tombstone iterator is required from an SST, and as a result it needs to be compatible with the `InternalIterator` interface.